File: src/transformers/features/http/HttpParameterTransformer.ts
import { HttpParameterProgrammer } from "../../../programmers/http/HttpParameterProgrammer";

import type { ITransformProps } from "../../ITransformProps";
import { GenericTransformer } from "../../internal/GenericTransformer";

export namespace HttpParameterTransformer {
  export const transform = (props: ITransformProps) =>
    GenericTransformer.scalar({
      ...props,
      method: "http.parameter",
      write: HttpParameterProgrammer.write,
    });
}

File: packages/typia/native/transform/features/http/HttpParameterTransformer.go
package http

import (
	shimast "github.com/microsoft/typescript-go/shim/ast"
	nativecontext "github.com/samchon/typia/packages/typia/native/core/context"
	nativehttpprogrammers "github.com/samchon/typia/packages/typia/native/core/programmers/http"
	nativeinternal "github.com/samchon/typia/packages/typia/native/transform/internal"
)

type httpParameterTransformerNamespace struct{}

var HttpParameterTransformer = httpParameterTransformerNamespace{}

func (httpParameterTransformerNamespace) Transform(props nativeinternal.ITransformProps) *shimast.Node {
	return nativeinternal.GenericTransformer.Scalar(nativeinternal.GenericTransformer_IProps{
		ITransformProps: props,
		Method:          "http.parameter",
		Write: func(x nativecontext.IProgrammerProps) *shimast.Node {
			return nativehttpprogrammers.HttpParameterProgrammer.Write(x)
		},
	})
}
